tests/harness.ps1

```
@@ -355,6 +355,22 @@ Write-Result 'batch.req014.consent' 'REQ-014: system Python consent gate log lin
+# reaching real console output). Fixed to the one-character form the sibling HP_CRUMB_FILE guard
+# already uses correctly. Assert the fixed form is present and the buggy two-backslash comparison
+# is gone, so a future edit cannot silently reintroduce the always-true condition.
+$scriptRootFixedForm = 'if not "%HP_SCRIPT_ROOT:~-1%"=="\" set "HP_SCRIPT_ROOT=%HP_SCRIPT_ROOT%\"'
+$hasScriptRootFixedForm = $AllText.Contains($scriptRootFixedForm)
+$scriptRootBuggyForm = '"%HP_SCRIPT_ROOT:~-1%"=="\\"'
+$hasScriptRootBuggyForm = $AllText.Contains($scriptRootBuggyForm)
+$scriptRootGuardOk = $hasScriptRootFixedForm -and (-not $hasScriptRootBuggyForm)
+Write-Result 'batch.script_root.trailing_backslash' 'HP_SCRIPT_ROOT trailing-backslash guard uses the correct one-character comparison, not the always-true two-backslash form' $scriptRootGuardOk @{
+    hasScriptRootFixedForm = $hasScriptRootFixedForm
+    hasScriptRootBuggyForm = $hasScriptRootBuggyForm
+}
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Write-Result' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)